prog/vm/nexus.rb

```diff
@@ -466,8 +499,36 @@ def write_params_json
 
     vm.vm_host.sshable.cmd("sudo host/bin/setup-vm delete_net #{q_vm}")
 
+    hop_destroy_slice
+  end
+
+  label def destroy_slice
+    slice = vm.vm_host_slice
+
+    # Remove the VM before we destroy the slice
     final_clean_up
 
+    # Trigger the slice deletion if this is a
+    # dedicated slice for this VM or if this is
+    # a shared slice but not used by any other VM.
+    # We do not need to wait for this to complete
+    #
+    # We disable the slice to prevent another
+    # concurrent VM allocation from grabbing it
+    # while it is being destroyed
+    if slice
+      # :nocov:
+      updated = VmHostSlice.dataset
+        .where(id: slice.id)
```

Comment:
```suggestion
      updated = slice.this
```

Since you already have the slice.